src/main/java/org/prebid/server/floors/BasicPriceFloorEnforcer.java

```diff
@@ -0,0 +1,213 @@
+                                bid.getId(), price, floor)));
+
+                // TODO: create a record for analytics adapters to be aware of this rejection and reason
+
+                updatedBidderBids.remove(bidderBid);
+            }
+        }
+
+        if (bidderBids.size() == updatedBidderBids.size()) {
+            return auctionParticipation;
+        }
+
+        final BidderSeatBid bidderSeatBid = BidderSeatBid.of(updatedBidderBids, seatBid.getHttpCalls(), errors);
+        return auctionParticipation.with(bidderResponse.with(bidderSeatBid));
+    }
+
+    private static boolean isEnforcedDealFloors(ExtRequestPrebidFloors floors,
+                                                AccountPriceFloorsConfig accountPriceFloorsConfig) {
+
+        final ExtRequestPrebidFloorsEnforcement enforcement = ObjectUtil.getIfNotNull(floors,
+                ExtRequestPrebidFloors::getEnforcement);
+        final Boolean requestEnforceDealFloors = ObjectUtil.getIfNotNull(enforcement,
+                ExtRequestPrebidFloorsEnforcement::getFloorDeals);
+
+        final Boolean accountEnforceDealFloors = accountPriceFloorsConfig.getEnforceDealFloors();
+
+        return BooleanUtils.isTrue(requestEnforceDealFloors) && BooleanUtils.isTrue(accountEnforceDealFloors);
+    }
+
+    private BigDecimal resolveFloor(BidderBid bidderBid,
+                                    BidRequest bidderBidRequest,
+                                    BidRequest bidRequest) {
+
+        final PriceFloorInfo priceFloorInfo = bidderBid.getPriceFloorInfo();
+        final BigDecimal customBidderFloor = ObjectUtil.getIfNotNull(priceFloorInfo, PriceFloorInfo::getFloor);
+        if (customBidderFloor != null) {
+            return convertIfRequired(customBidderFloor, priceFloorInfo.getCurrency(), bidderBidRequest, bidRequest);
+        }
+
+        final Imp imp = correspondingImp(bidderBid.getBid(), bidderBidRequest.getImp());
+        return convertIfRequired(imp.getBidfloor(), imp.getBidfloorcur(), bidderBidRequest, bidRequest);
+    }
+
+    /**
+     * Converts floor according to incoming auction request currency (since response bid is already converted).
+     * <p>
+     * Floor currency resolved in order: floor currency -> bidder request currency -> auction request currency.
+     */
+    private BigDecimal convertIfRequired(BigDecimal floor,
+                                         String floorCurrency,
+                                         BidRequest bidderBidRequest,
+                                         BidRequest bidRequest) {
+
+        final String resolvedFloorCurrency = ObjectUtils.defaultIfNull(floorCurrency,
+                resolveBidRequestCurrency(bidderBidRequest));
+
+        final String bidRequestCurrency = resolveBidRequestCurrency(bidRequest);
+
+        if (resolvedFloorCurrency != null && !resolvedFloorCurrency.equals(bidRequestCurrency)) {
+            return currencyConversionService.convertCurrency(
```

Comment:
Should we handle exceptions from this method?

Reply:
Good catch, thx!